src/charm.py

```diff
@@ -100,46 +95,88 @@ def _on_install(self, event: EventBase) -> None:
             return
         self._generate_home_network_private_key()
 
-    def _configure_sdcore_udm(self, event: EventBase) -> None:
-        """Adds Pebble layer and manages Juju unit status.
-
-        Args:
-            event (EventBase): Juju event.
-        """
+    def ready_to_configure(self) -> bool:
```

Comment:
I guess I have the same comments as in the NSSF one, so let's agree over there before continuing this one here.